sds_data_manager/utils/stackbuilder.py

```diff
@@ -47,112 +47,135 @@ def build_sds(
         Account configuration (domain_name and other account specific configurations)
 
     """
-    data_bucket = data_bucket_stack.DataBucketStack(
-        scope=scope, construct_id="DataBucket", env=env
+    networking_stack = Stack(scope, "NetworkingStack", env=env)
+    networking = networking_construct.NetworkingConstruct(
+        networking_stack, "Networking"
     )
 
-    networking = networking_stack.NetworkingStack(scope, "Networking", env=env)
+    sdc_stack = Stack(scope, "SDCStack", env=env)
 
-    monitoring = monitoring_stack.MonitoringStack(
-        scope=scope,
-        construct_id="MonitoringStack",
-        env=env,
+    # Adding this endpoint so that lambda within
+    # this VPC can perform boto3.client("events")
+    # or boto3.client("batch") operations
+    networking.vpc.add_interface_endpoint(
+        "EventBridgeEndpoint",
+        service=ec2.InterfaceVpcEndpointAwsService.EVENTBRIDGE,
+    )
+    networking.vpc.add_interface_endpoint(
+        "BatchJobEndpoint", service=ec2.InterfaceVpcEndpointAwsService.BATCH
+    )
+
+    # The lambda is in the same private security group as the RDS, but
+    # it needs to access the secrets manager, so we add this endpoint.
+    networking.vpc.add_interface_endpoint(
+        "SecretManagerEndpoint",
+        service=ec2.InterfaceVpcEndpointAwsService.SECRETS_MANAGER,
+        subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
+        private_dns_enabled=True,
+    )
+
+    data_bucket = data_bucket_construct.DataBucketConstruct(
+        scope=sdc_stack, construct_id="DataBucket", env=env
+    )
+
+    monitoring = monitoring_construct.MonitoringConstruct(
+        scope=sdc_stack,
+        construct_id="MonitoringConstruct",
     )
 
     domain = None
     domain_name = account_config.get("domain_name", None)
     account_name = account_config["account_name"]
     if domain_name is not None:
-        domain = domain_stack.DomainStack(
-            scope,
-            "DomainStack",
+        domain = domain_construct.DomainConstruct(
+            scope=sdc_stack,
+            construct_id="DomainConstruct",
             domain_name=domain_name,
             account_name=account_name,
-            env=env,
         )
 
-    api = api_gateway_stack.ApiGateway(
-        scope,
-        "ApiGateway",
-        domain_stack=domain,
-        env=env,
+    api = api_gateway_construct.ApiGateway(
+        scope=sdc_stack,
+        construct_id="ApiGateway",
+        domain_construct=domain,
     )
     api.deliver_to_sns(monitoring.sns_topic_notifications)
 
+    # create Code asset and Layer for Lambda(s)
+    layer_code_directory = (
+        Path(__file__).parent.parent.parent / "lambda_layer/python"
+    ).resolve()
+    lambda_code_directory = Path(__file__).parent.parent / "lambda_code"
+
+    lambda_code = lambda_.Code.from_asset(str(lambda_code_directory))
+    db_lambda_layer = lambda_layer_construct.LambdaLayerConstruct(
```

Comment:
Why do we only have a single lambda layer and it's named for the database?

Reply:
We want to build the dependencies only once (@tech3371 's previous PR), and attach that to all the Lambda Functions that need it. Here this is the database requirements layer. For instance, if you add "imap_processing" as a requirement for the SPICE kernel work, that could be a separate layer and only attached to the Lambda Functions that need that specific piece.

Reply:
Thanks for the explanation. I'm still getting intuition on how this works.